### PR TITLE
WFLY-15562 Simplify logging and validation in various createTimer met…

### PR DESCRIPTION
**Pull request by chengfang**
…hods in TimerServiceImpl

https://issues.redhat.com/browse/WFLY-15562

### CODE REVIEW

**Review by tadamski — APPROVED** [review]
LGTM

File: ejb3/src/main/java/org/jboss/as/ejb3/timerservice/TimerServiceImpl.java [hunk 1]
```diff
@@ -257,13 +260,13 @@ public Timer createIntervalTimer(Date initialExpiration, long intervalDuration,
             throws IllegalArgumentException, IllegalStateException, EJBException {
         assertTimerServiceState();
         if (initialExpiration == null) {
-            throw EJB3_TIMER_LOGGER.initialExpirationIsNullCreatingTimer();
+            throw EJB3_TIMER_LOGGER.invalidTimerParameter("initialExpiration", null);
```
Comment (boris-unckel) [inline ejb3/src/main/java/org/jboss/as/ejb3/timerservice/TimerServiceImpl.java:263]: You might consider [org.wildfly.common.Assert.checkNotNullParam](https://github.com/wildfly/wildfly-common/blob/1.5.4.Final/src/main/java/org/wildfly/common/Assert.java#L47) for these cases.

File: ejb3/src/main/java/org/jboss/as/ejb3/timerservice/TimerServiceImpl.java [hunk 2]
```diff
@@ -257,13 +260,13 @@ public Timer createIntervalTimer(Date initialExpiration, long intervalDuration,
...
         }
         if (initialExpiration.getTime() < 0) {
```
Comment (boris-unckel) [inline ejb3/src/main/java/org/jboss/as/ejb3/timerservice/TimerServiceImpl.java:265]: You might consider [org.wildfly.common.Assert.checkMinimumParameter](https://github.com/wildfly/wildfly-common/blob/1.5.4.Final/src/main/java/org/wildfly/common/Assert.java#L305) for these cases.

**Comment by chengfang** [comment]
While reusing WildFly common Assert is good, using ejb's own error logging allows for more direct handling and targeted message. So in this case, I'm leaning towards ejb's own.

File: ejb3/src/main/java/org/jboss/as/ejb3/timerservice/TimerServiceImpl.java (hunk 1 above)
Comment (bstansberry) [reply ejb3/src/main/java/org/jboss/as/ejb3/timerservice/TimerServiceImpl.java:263]: The EJB3_TIMER_LOGGER message provides more context and is internationalized.

**Pull request merged**